src/libs/EmojiUtils.ts

```diff
@@ -355,26 +357,31 @@ function replaceEmojis(text: string, preferredSkinTone = CONST.EMOJI_DEFAULT_SKI
 
             // If this is the last emoji in the message and it's the end of the message so far,
             // add a space after it so the user can keep typing easily.
-            if (i === emojiData.length - 1) {
+            if (newText.indexOf(emoji) + emoji.length === newText.length) {
```

Comment:
This change is not really related to the issue this PR is addressing but there are a few problems I found with how it was handling adding spaces at the end of strings:
* If you paste "abc :joy: 123" it would add an extra space after 😂, yet cursor would be positioned after 123 (correct behavior)
* if you paste text with invalid emojis, it would not add space. For example: "abc :joy: 123 :invalidemoji: 321". The original intention was to add a space after 😂 but in this case it would not because it would think :joy: was not the last emoji because there's one more after it, even though its an invalid one.

The change above will add a space ONLY if it's the last emoji and there's nothing after it. I'm not sure this will change any expected behavior (there's a test that was testing for that behavior, which I had to change), so I can revert this back if you think that's the best thing to go (and this could be revisited in another PR).

Reply:
Got it, I suspected I would be missing something. As this is not really relevant for the issue at hand, I'll just revert back to the original behavior. It will still be flawed under some scenarios (described above) but at least we'll be back to the original behavior and this can be dealt in a separate issue. https://github.com/Expensify/App/pull/30412 deals with something similar and I left a couple of comments on that issue with ideas to make it more generic when it comes to add space after emojis.

Reply:
@Ollyws I pushed an update to it that I believe is a good solution to keep the current behavior while fixing a couple of issues I found with the original approach. 
Here's what I tried to solve. Go to staging and try the following combinations:

* `Abc :joy:123 :invalid:` it will NOT add space after 😂 because the logic assumes all emoji codes found is valid, which is not true.
* `Abc :joy: 123`: it will add a second space after 😂 because it blindly add space after the last emoji.
* `Abc :joy: ` (note the trailing space) same as above, double space at the end.
* type `abc 123` then `:joy:` right after c (before the space). It will add another space

The [changes in this PR](https://github.com/Expensify/App/pull/31279/files#diff-1692fc6ebb559e5fd701e993d1e71ebee2b6ed378c84d2a04654b3ff3ff20c50R367-R376) will address these scenarios (I added tests to validate it). it will correctly find the last emoji and also it will not add space if there's already a space after the emoji, while still positioning the cursor after the space (4th item in the list above), which is the expected behavior IMO.

I still think the whole "adding space after emoji" needs to be revisited because we have similar logic in multiple places and I believe it can be centralized, but this deserves it's own issue (possibly part of #30412)